### PR TITLE
New Klarna Methods Payone

### PR DESCRIPTION
**Pull request by cmanciu-plenty**
### Requirements that must be fulfilled:
- [ ] New version in `plugin.json` updated
- [ ] Changelog entry added
- [X ] Plugin can be built (`build-set`)
- [X] Tested by the author
- [ ] Tested by a second person

[Kanbanize Card](LINK)

@plentymarkets/team-order-payment
____________________
New version must be uploaded from PID 31920.

Further information about the plugin can be found in the [Wiki](https://wiki.plentymarkets.com/display/OP/Payone).

### CODE REVIEW

File: src/Models/Settings.php [hunk 1]
```diff
@@ -148,6 +151,18 @@ public function updateValues(array $data): Model
         if (isset($data['PAYONE_PAYONE_AMAZON_PAY'])) {
             $this->value['PAYONE_PAYONE_AMAZON_PAY'] = $data['PAYONE_PAYONE_AMAZON_PAY'];
         }
+        if (isset($data['PAYONE_PAYONE_AMAZON_PAY'])) {
```
Comment (j-konopka) [inline src/Models/Settings.php]: Copy & Paste fail ;)

File: src/Models/Settings.php [hunk 2]
```diff
@@ -148,6 +151,18 @@ public function updateValues(array $data): Model
...
+            $this->value['PAYONE_PAYONE_KLARNA_DIRECT_BANK'] = $data['PAYONE_PAYONE_KLARNA_DIRECT_BANK'];
+        }
+        if (isset($data['PAYONE_PAYONE_KLARNA_DIRECT_DEBIT'])) {
+            $this->value['PAYONE_PAYONE_KLARNA_DIRECT_DEBIT'] = $data['PAYONE_PAYONE_KLARNA_DIRECT_DEBIT'];
+        }
+        if (isset($data['PAYONE_PAYONE_KLARNA_INSTALLMENTS'])) {
+            $this->value['PAYONE_PAYONE_KLARNA_INSTALLMENTS'] = $data['PAYONE_PAYONE_KLARNA_INSTALLMENTS'];
+        }
+        if (isset($data['PAYONE_PAYONE_AMAZON_PAY'])) {
```
Comment (j-konopka) [inline src/Models/Settings.php]: Copy & Paste fail ;)

**Review by j-konopka — CHANGES_REQUESTED** [review]
See comments.

The migration to create the payment methods in the database is missing.

**Comment by cmanciu-plenty** [comment]
Added Migration Also

**Review by j-konopka — APPROVED** [review]

**Review by MKanitz — APPROVED** [review]

**Pull request merged**